state/machine.go

```diff
@@ -2210,483 +2208,6 @@ func (m *Machine) VerifyUnitsSeries(unitNames []string, series string, force boo
 	return results, nil
 }
 
-// instanceCharmProfileDataC holds attributes relevant to a lxd charm profile
-// data that's on the machine
-type instanceCharmProfileData struct {
```

Comment:
Does there need to be an upgrade step to delete any existing data that we now no longer use?

Reply:
Any docs in this collection are transient and "should" not exist at the time of an upgrade.  However an upgrade step won't hurt, to be sure.  Will add.